{
    "critiques": {
        "700": {
            "general_comment": null,
            "rubric_items": {
                "7964": {
                    "autograded": false,
                    "observation": "Correct!",
                    "result": "passed"
                },
                "5572": {
                    "autograded": false,
                    "observation": "Correct!",
                    "result": "passed"
                },
                "5570": {
                    "autograded": false,
                    "observation": "```\nNot all of the tests passed.\n\n\n************************************************************************\n                         Test Failure Feedback                          \n************************************************************************\n\nFailed Test: The weights are updated correctly on training\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"/home/grader/first_neural_network/workspace/my_answers.py\", line 53, in train\n    self.update_weights(delta_weights_i_h, delta_weights_h_o, n_records)\n  File \"/home/grader/first_neural_network/workspace/my_answers.py\", line 119, in update_weights\n    self.weights_hidden_to_output += self.lr*delta_weights_h_o/self.n_records\nAttributeError: 'NeuralNetwork' object has no attribute 'n_records'\n\n\n************************************************************************\n                          Test Result Summary                           \n************************************************************************\n\nThe weights are updated correctly on training                          E\n\n\n```",
                    "result": "failed"
                },
                "5573": {
                    "autograded": false,
                    "observation": "Correct!",
                    "result": "passed"
                },
                "7965": {
                    "autograded": false,
                    "observation": "```\nNot all of the tests passed.\n\n\n************************************************************************\n                         Test Failure Feedback                          \n************************************************************************\n\nFailed Test: Produces good results when running the network on full data\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"/home/grader/first_neural_network/workspace/my_answers.py\", line 53, in train\n    self.update_weights(delta_weights_i_h, delta_weights_h_o, n_records)\n  File \"/home/grader/first_neural_network/workspace/my_answers.py\", line 119, in update_weights\n    self.weights_hidden_to_output += self.lr*delta_weights_h_o/self.n_records\nAttributeError: 'NeuralNetwork' object has no attribute 'n_records'\n\n\n************************************************************************\n                          Test Result Summary                           \n************************************************************************\n\nProduces good results when running the network on full data            E\n\n\n```",
                    "result": "failed"
                },
                "5578": {
                    "autograded": false,
                    "observation": "Correct!",
                    "result": "passed"
                },
                "5582": {
                    "autograded": false,
                    "observation": "Correct!",
                    "result": "passed"
                },
                "5575": {
                    "autograded": false,
                    "observation": "Correct!",
                    "result": "passed"
                },
                "5581": {
                    "autograded": false,
                    "observation": "Correct!",
                    "result": "passed"
                },
                "5574": {
                    "autograded": false,
                    "observation": "Correct!",
                    "result": "passed"
                },
                "5580": {
                    "autograded": false,
                    "observation": "```\nNot all of the tests passed.\n\n\n************************************************************************\n                         Test Failure Feedback                          \n************************************************************************\n\nFailed Test: The update_weights implementation is correct\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"/home/grader/first_neural_network/workspace/my_answers.py\", line 119, in update_weights\n    self.weights_hidden_to_output += self.lr*delta_weights_h_o/self.n_records\nAttributeError: 'CorrectNeuralNetwork' object has no attribute 'n_records'\n\n\n************************************************************************\n                          Test Result Summary                           \n************************************************************************\n\nThe update_weights implementation is correct                           E\n\n\n```",
                    "result": "failed"
                }
            }
        }
    },
    "tests": [
        {
            "traceback": "Traceback (most recent call last):\n  File \"/home/grader/first_neural_network/workspace/my_answers.py\", line 53, in train\n    self.update_weights(delta_weights_i_h, delta_weights_h_o, n_records)\n  File \"/home/grader/first_neural_network/workspace/my_answers.py\", line 119, in update_weights\n    self.weights_hidden_to_output += self.lr*delta_weights_h_o/self.n_records\nAttributeError: 'NeuralNetwork' object has no attribute 'n_records'\n",
            "rubric_item_id": 7965,
            "description": "Produces good results when running the network on full data",
            "result": "E"
        },
        {
            "traceback": "Traceback (most recent call last):\n  File \"/home/grader/first_neural_network/workspace/my_answers.py\", line 53, in train\n    self.update_weights(delta_weights_i_h, delta_weights_h_o, n_records)\n  File \"/home/grader/first_neural_network/workspace/my_answers.py\", line 119, in update_weights\n    self.weights_hidden_to_output += self.lr*delta_weights_h_o/self.n_records\nAttributeError: 'NeuralNetwork' object has no attribute 'n_records'\n",
            "rubric_item_id": 5570,
            "description": "The weights are updated correctly on training",
            "result": "E"
        },
        {
            "traceback": "Traceback (most recent call last):\n  File \"/home/grader/first_neural_network/workspace/my_answers.py\", line 119, in update_weights\n    self.weights_hidden_to_output += self.lr*delta_weights_h_o/self.n_records\nAttributeError: 'CorrectNeuralNetwork' object has no attribute 'n_records'\n",
            "rubric_item_id": 5580,
            "description": "The update_weights implementation is correct",
            "result": "E"
        },
        {
            "traceback": "",
            "rubric_item_id": 5574,
            "description": "The activation function is a sigmoid",
            "result": "."
        },
        {
            "traceback": "",
            "rubric_item_id": 5578,
            "description": "The backpropagation implementation is correct",
            "result": "."
        },
        {
            "traceback": "",
            "rubric_item_id": 5575,
            "description": "The number of epochs is reasonable",
            "result": "."
        },
        {
            "traceback": "",
            "rubric_item_id": 5572,
            "description": "The forward pass implementation is correct",
            "result": "."
        },
        {
            "traceback": "",
            "rubric_item_id": 5581,
            "description": "The number of hidden nodes is reasonable",
            "result": "."
        },
        {
            "traceback": "",
            "rubric_item_id": 5582,
            "description": "The learning_rate is reasonable",
            "result": "."
        },
        {
            "traceback": "",
            "rubric_item_id": 7964,
            "description": "The number of output nodes is correct",
            "result": "."
        },
        {
            "traceback": "",
            "rubric_item_id": 5573,
            "description": "The run method is correct",
            "result": "."
        }
    ]
}